### ISSUE TITLE
Include features from GooEngine Blender fork

### ISSUE THREAD

**Issue by ghost** [Bforartists/Bforartists#3683]
- [x] Add Toggle for Custom Line Width in Viewport
https://github.com/dillongoostudios/goo-engine/commit/abba7abadcedcaa77d8a5cfddef1c619719867ea

- [x] #3688
https://github.com/dillongoostudios/goo-engine/commit/d5e668d1cf650e9551478e1d1121e82298baa8e7

- [x] Copy Rotation constraint: Add Invert all option. 
https://github.com/dillongoostudios/goo-engine/commit/1c9eb790e0b3351b0df2d27d67bdc80e5620573c

- [ ] Add pose constraint merge operator 
https://github.com/dillongoostudios/goo-engine/commit/963923b286024f908a0ca2afd70057ebfabfd9cb

- [x] Show FCurve in editor operator 
https://github.com/dillongoostudios/goo-engine/commit/1574a31c15022dfa87e669c7b2734cd939858a71

- [ ] Add 'Hide in Outliner' option to pose bones and new outliner
https://github.com/dillongoostudios/goo-engine/commit/f6917850f6243ca4b1c6157d275b5a42adf95a3a

- [ invalid, has gone upstream into Blender already] Enable GP screenspace thickness slider for all modes. 
https://github.com/dillongoostudios/goo-engine/commit/7cefd01aaf2e2741f8a79df7060ddf18d811c6bc

- [x] GooEngine: Option to disable Material Icon rendering. 
https://github.com/dillongoostudios/goo-engine/commit/59378aace1e07051c553d9197304a1de4b80bcce

**Comment by Draise14** [comment]
Yeah let's go.  I talked with one of the devs, the GUI and UX tasks is ok for cross-pollination. The revert engine they are doing is somewhat a breaking change so we will have to live without for now -but these would be handy! 

My only note, instead of one big commit - I think one task per new feature night make it easier to work the documentation so this can be added in bite sized ways and be somewhat tested per patch.

**Comment by ReinerBforartists** [comment]
Yes, breakt it up into smaller tasks please.

**Comment by iyadahmed** [comment]
ok gonna scrap all the work :(

**Comment by ReinerBforartists** [comment]
Awww :(

The reason to separate as much as possible is that in the future the single tasks should be followable. And this doesn't work when everything is in one big commit. 

Let's say we have a change in the "Enable GP screenspace thickness slider for all modes" task. And need to search for how it was implemented in the first way. And then we stumble across no commit since it was part of a bunch of commits. And even when we might find it, it is still burried in the heap.

That's why it is always better to separate the tasks as much as possible. Looks also much better in the list of commits then. The user can see several separate tasks then, and can follow what exactly was improved.

**Comment by iyadahmed** [comment]
No worries :sparkles:

**Comment by Draise14** [comment]
Postponed, will add more of these in 4.1

**Comment by ReinerBforartists** [comment]
I'll try the copy rotate constraint improvement. Looks easy enough.

**Comment by ReinerBforartists** [comment]
Remaining four tasks are not to solve, at least not for me. I would let go ...

**Comment by Draise14** [comment]
Will close.
